src/internal/error/errors.go

```diff
@@ -0,0 +1,147 @@
+package error
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"strings"
+)
+
+// Error ...
+type Error struct {
+	Cause   error  `json:"-"`
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+// Error returns a human readable error.
+func (e *Error) Error() string {
+	return fmt.Sprintf("%v, %s, %s", e.Cause, e.Code, e.Message)
+}
+
+// WithMessage ...
+func (e *Error) WithMessage(msg string) *Error {
+	e.Message = msg
+	return e
+}
+
+// WithCode ...
+func (e *Error) WithCode(code string) *Error {
+	e.Code = code
+	return e
+}
+
+// Unwrap ...
+func (e *Error) Unwrap() error { return e.Cause }
+
+// Errors ...
+type Errors []error
+
+var _ error = Errors{}
+
+// Error converts slice of error
+func (errs Errors) Error() string {
+	var tmpErrs struct {
+		Errors []Error `json:"errors,omitempty"`
+	}
+
+	for _, e := range errs {
+		var err error
+		switch e.(type) {
+		case *Error:
+			err = e.(*Error)
+		default:
+			err = UnknownError(e).WithMessage(err.Error())
+		}
+		tmpErrs.Errors = append(tmpErrs.Errors, *err.(*Error))
+	}
+
+	msg, err := json.Marshal(tmpErrs)
+	if err != nil {
+		log.Error(err)
+		return "{}"
+	}
+	return string(msg)
+}
+
+// Len returns the current number of errors.
+func (errs Errors) Len() int {
+	return len(errs)
+}
+
+// NewErrs ...
+func NewErrs(err error) Errors {
+	return Errors{err}
+}
+
+const (
+	// NotFoundCode is code for the error of no object found
+	NotFoundCode = "NOT_FOUND"
+	// ConflictCode ...
+	ConflictCode = "CONFLICT"
+	// UnAuthorizedCode ...
+	UnAuthorizedCode = "UNAUTHORIZED"
+	// BadRequestCode ...
+	BadRequestCode = "BAD_REQUEST"
+	// ForbiddenCode ...
+	ForbiddenCode = "FORBIDDER"
```

Comment:
FORBIDDEN?

Reply:
Fix included in https://github.com/goharbor/harbor/pull/10366